.github/workflows/build_wheels.yml

```diff
@@ -0,0 +1,89 @@
+name: Build and upload to PyPI
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+  release:
+    types:
+      - published
+
+jobs:
+  build_wheels:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-22.04, windows-2022, macos-11]
```

Comment:
is there a reason for these particular choices?

Reply:
Nope, just 3 platforms that cibuildwheel supports. It's the platforms rather than the OS versions that matter